presto-blackhole/src/main/java/com/facebook/presto/plugin/blackhole/BlackHoleMetadata.java

```diff
@@ -0,0 +1,218 @@
+        return tables.values().stream()
+                .map(BlackHoleTableHandle::toSchemaTableName)
+                .collect(toList());
+    }
+
+    @Override
+    public ColumnHandle getSampleWeightColumnHandle(ConnectorTableHandle tableHandle)
+    {
+        //  returns null as the table does not contain sampled data
+        // (see {@link com.facebook.presto.spi.ConnectorMetadata.getSampleWeightColumnHandle()}
+        return null;
+    }
+
+    @Override
+    public boolean canCreateSampledTables(ConnectorSession session)
+    {
+        return false;
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorTableHandle tableHandle)
+    {
+        BlackHoleTableHandle blackHoleTableHandle = checkType(tableHandle, BlackHoleTableHandle.class, "tableHandle");
+        return blackHoleTableHandle.getColumnHandles().stream()
+                .collect(toMap(BlackHoleColumnHandle::getName, column -> column));
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        BlackHoleColumnHandle blackHoleColumnHandle = checkType(columnHandle, BlackHoleColumnHandle.class, "columnHandle");
+        return blackHoleColumnHandle.toColumnMetadata(typeManager);
+    }
+
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        return tables.values().stream()
+                .filter(table -> prefix.matches(table.toSchemaTableName()))
+                .collect(toMap(BlackHoleTableHandle::toSchemaTableName, handle -> handle.toTableMetadata(typeManager).getColumns()));
+    }
+
+    @Override
+    public void dropTable(ConnectorTableHandle tableHandle)
+    {
+        BlackHoleTableHandle blackHoleTableHandle = checkType(tableHandle, BlackHoleTableHandle.class, "tableHandle");
+        tables.remove(blackHoleTableHandle.getTableName());
+    }
+
+    @Override
+    public void renameTable(ConnectorTableHandle tableHandle, SchemaTableName newTableName)
+    {
+        BlackHoleTableHandle oldTableHandle = checkType(tableHandle, BlackHoleTableHandle.class, "tableHandle");
+        BlackHoleTableHandle newTableHandle = new BlackHoleTableHandle(
+                oldTableHandle.getSchemaName(), newTableName.getTableName(), oldTableHandle.getColumnHandles());
+        synchronized (tables) {
+            tables.remove(oldTableHandle.getTableName());
+            tables.put(newTableName.getTableName(), newTableHandle);
+        }
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        beginCreateTable(session, tableMetadata);
+    }
+
+    @Override
+    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        tables.put(tableMetadata.getTable().getTableName(), new BlackHoleTableHandle(tableMetadata));
```

Comment:
This should be done in commit, not begin, since the query that reads data for the table could fail (in which case the table should not be created)